src/frontend/js/widgets/Dashboard/components/DashboardItem/Order/OrderStateMessage/index.tsx

```diff
@@ -0,0 +1,92 @@
+import { captureException } from '@sentry/browser';
+import { FormattedMessage } from 'react-intl';
+
+import { Order, OrderState } from 'types/Joanie';
+import { StringHelper } from 'utils/StringHelper';
+
+export const messages = {
+  statusDraft: {
+    id: 'components.DashboardItemOrder.statusDraft',
+    description: 'Status shown on the dashboard order item when order is draft.',
+    defaultMessage: 'Draft',
+  },
+  statusSubmitted: {
+    id: 'components.DashboardItemOrder.statusSubmitted',
+    description: 'Status shown on the dashboard order item when order is submitted.',
+    defaultMessage: 'Submitted',
+  },
+  statusPending: {
+    id: 'components.DashboardItemOrder.statusPending',
+    description: 'Status shown on the dashboard order item when order is pending.',
+    defaultMessage: 'Pending',
+  },
+  statusOnGoing: {
+    id: 'components.DashboardItemOrder.statusOnGoing',
+    description:
+      'Status shown on the dashboard order item when order is validated with no certificate',
+    defaultMessage: 'On going',
+  },
+  statusCompleted: {
+    id: 'components.DashboardItemOrder.statusCompleted',
+    description:
+      'Status shown on the dashboard order item when order is validated with certificate',
+    defaultMessage: 'Completed',
+  },
+  statusWaitingSignature: {
+    id: 'components.DashboardItemOrder.statusWaitingSignature',
+    description:
+      "Status shown on the dashboard order item when order is validated with contract's learner signature missing.",
+    defaultMessage: 'Signature required',
+  },
+  statusCanceled: {
+    id: 'components.DashboardItemOrder.statusCanceled',
+    description: 'Status shown on the dashboard order item when order is canceled',
+    defaultMessage: 'Canceled',
+  },
+  statusOther: {
+    id: 'components.DashboardItemOrder.statusOther',
+    description: 'Status shown on the dashboard order item when order status is unknown',
+    defaultMessage: '{state}',
+  },
+};
+
+interface OrderStateMessageProps {
+  order: Order;
+}
+
+const OrderStateMessage = ({ order }: OrderStateMessageProps) => {
+  const { contract, certificate } = order;
+
+  if (order.state === OrderState.VALIDATED) {
+    if (contract && !contract.sign_date) {
+      return <FormattedMessage {...messages.statusWaitingSignature} />;
+    }
+
+    if (certificate) {
+      return <FormattedMessage {...messages.statusCompleted} />;
+    } else {
+      return <FormattedMessage {...messages.statusOnGoing} />;
+    }
+  }
+
+  const orderStatusMessages = {
+    [OrderState.DRAFT]: messages.statusDraft,
+    [OrderState.SUBMITTED]: messages.statusSubmitted,
+    [OrderState.PENDING]: messages.statusPending,
+    [OrderState.CANCELED]: messages.statusCanceled,
+  };
+  if (order.state in orderStatusMessages) {
+    return <FormattedMessage {...orderStatusMessages[order.state]} />;
+  }
+
+  captureException(new Error(`Unknow order state ${order.state}`));
```

Comment:
You should use the `handle` method instead. (`utils/errors/handle.ts`).

Then you should invoke this method within a useEffect to prevent to spam sentry or console at each render.